Draft Common Message Queue

vikramkoka opened the pull request:
Here is a very early draft PR to introduce and socialize the concept of a "common message queue" abstraction similar to the "Common SQL" and "Common IO" abstractions in Airflow.

This will be a provider package similar to those and is intended to be an abstraction over Apache Kafka, Amazon SQL, and Google PubSub to begin with. It can then be expanded to other messaging systems based on community adoption.

The initial goal with this is to provide a simple abstraction for integrating Event Driven Scheduling coming with Airflow 3 to message notification systems such as Kafka, currently being used to publish data availability.

At this stage, this is very much a WIP draft intended to solicit input from the community.

vikramkoka committed:
Draft Common Message Queue

Here is a very early draft PR to introduce and socialize the concept of a "common message queue" abstraction similar to the "Common SQL" and "Common IO" abstractions in Airflow.

This will be a provider package similar to those and is intended to be an abstraction over Apache Kafka, Amazon SQL, and Google PubSub to begin with. It can then be expanded to other messaging systems based on community adoption.

The initial goal with this is to provide a simple abstraction for integrating Event Driven Scheduling coming with Airflow 3 to message notification systems such as Kafka, currently being used to publish data availability.

At this stage, this is very much a WIP draft intended to solicit input from the community.

vikramkoka committed:
Updated the Readme for CommonMsgQueue

Updated the Common Message Queue Readme with an example of an Event Driven Dag

vikramkoka committed:
Updated the message operator and sensor

Updated the message queue Operator and Sensor to fix an issue in my sync

vikramkoka committed:
Changed Sensor to Trigger

Changed the Message Queue Sensor Operator to be a Deferrable Trigger

vikramkoka committed:
Fixed typos and imports

Fixed typos and import errors in the MsgQueueHook

shubham22 commented on providers/common/msgq/README.rst:
```diff
@@ -0,0 +1,101 @@
+
+Expected usage would be something on the lines below::
+   
+   trigger = = MsgQueueSensorTrigger(msg_queue="https://sqs.us-east-1.amazonaws.com/722404908466/Test")
```
I believe this should be `MessageQueueTrigger(msg_queue...)` as the trigger class name is MessageQueueTrigger.

vikramkoka replied:
Yes, you are right. Thanks for pointing that out.

vikramkoka committed:
Updated README to fix typo

Updated invocation of MsqQueueSensorTrigger to MsgQueueTrigger in example invocation

jscheffl commented on providers/common/msgq/README.rst:
```diff
@@ -0,0 +1,101 @@
+
+Expected usage would be something on the lines below::
+   
+   trigger = = MsgQueueTrigger(msg_queue="https://sqs.us-east-1.amazonaws.com/722404908466/Test")
```
Typo?
```suggestion
   trigger = MsgQueueTrigger(msg_queue="https://sqs.us-east-1.amazonaws.com/722404908466/Test")
```

jscheffl commented on providers/common/msgq/README.rst:
```diff
@@ -0,0 +1,101 @@
+   
+   trigger = = MsgQueueTrigger(msg_queue="https://sqs.us-east-1.amazonaws.com/722404908466/Test")
+   
+   data_asset = Asset("incoming_asset", watchers=[
```
Spacing
```suggestion
    data_asset = Asset("incoming_asset", watchers=[
```

jscheffl commented on providers/common/msgq/src/airflow/providers/common/msgq/operators/msg_queue.py:
```diff
@@ -0,0 +1,109 @@
+    "amazon": "4.1.0",
+    "apache.kafka": "2.1.0",
+    "google": "8.2.0",
+}
```
I am with Vincent here - we should not have a reverse dependency to the providers using this feature.

jscheffl reviewed (commented):
In general looks good. Some more nit that I would have on the Python code/Interface but we can leave this until it is in real review.

Would be great to add an example DAG as well for the showcase.

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Rename `msgq` to `messaging`

vincbeck committed:
Implement `MessageQueueTrigger`

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Create provider files + create base class for providers

vincbeck committed:
Add unit tests

vincbeck committed:
Create doc and system test

vincbeck committed:
Add doc to describe providers

vincbeck committed:
Fix test

jedcunningham commented on providers/common/messaging/docs/changelog.rst:
```diff
@@ -0,0 +1,35 @@
+.....
+
+.. note::
+  This release of provider is only available for Airflow 2.9+ as explained in the
```
This doesn't match the 3.0+ requirement.

jedcunningham commented on providers/common/messaging/tests/system/common/messaging/example_message_queue_trigger.py:
```diff
@@ -0,0 +1,37 @@
+):
+    task = EmptyOperator(task_id="task")
+
+    chain(task)
```
```suggestion
    EmptyOperator(task_id="task")
```

No need for a var and chain if there is a single task.

jedcunningham commented on providers/common/messaging/tests/system/common/messaging/example_message_queue_trigger.py:
This feels a little weird being in the test dir?

vincbeck replied:
Why? It is a system test. All system tests are in tests directory

jedcunningham replied:
Fair enough, I guess I spend significantly more time looking at core example dags than provider ones.

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Add missing init files

vincbeck committed:
Update providers/common/messaging/tests/system/common/messaging/example_message_queue_trigger.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

vincbeck committed:
Fix changelog

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Update `extra-packages-ref`

vincbeck committed:
Fix static checks

vincbeck committed:
Update `commits.rst`

vincbeck committed:
Exclude `common.messaging` from compat tests

vincbeck committed:
Add `common.messaging` to `airflow_providers_bug_report`

vincbeck committed:
Fix compat tests

vincbeck committed:
Fix doc

vincbeck committed:
Fix doc

vincbeck committed:
Fix doc

vincbeck committed:
Fix mypy

vincbeck committed:
Fix doc

vincbeck committed:
Merge branch 'main' into common-msgQ

vincbeck committed:
Add link

vincbeck committed:
Add modules to `exclude_patterns`
